daemon/cleanup.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2018 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/cilium/cilium/pkg/pidfile"
+)
+
+func handleInterrupt() <-chan struct{} {
+	// Handle the handleOSSignals
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, syscall.SIGQUIT, syscall.SIGINT, syscall.SIGHUP, syscall.SIGTERM)
+	interrupt := make(chan struct{})
+	go func() {
+		for s := range sig {
+			log.WithField("signal", s).Info("Exiting due to signal")
+			<-pidfile.Clean()
+			break
```

Comment:
We no longer do an `os.Exit(0)`, what is replacing it?

Reply:
@aanm how does `RootCmd.Execute()` return?